spec/std/process_spec.cr

```diff
@@ -3,54 +3,110 @@ require "process"
 require "./spec_helper"
 require "../spec_helper"
 
+private def exit_code_command(code)
+  {% if flag?(:win32) %}
+    {"cmd.exe", {"/c", "exit #{code}"}}
+  {% else %}
+    case code
+    when 0
+      {"true", [] of String}
+    when 1
+      {"false", [] of String}
+    else
+      {"/bin/sh", {"-c", "exit #{code}"}}
+    end
+  {% end %}
+end
+
+private def shell_command(command)
+  {% if flag?(:win32) %}
+    {"cmd.exe", {"/c", command}}
+  {% else %}
+    {"/bin/sh", {"-c", command}}
+  {% end %}
+end
+
+private def stdin_to_stdout_command
+  {% if flag?(:win32) %}
+    {"powershell.exe", {"-C", "$Input"}}
+  {% else %}
+    {"/bin/cat", [] of String}
+  {% end %}
+end
+
+private def standing_command
+  {% if flag?(:win32) %}
+    {"cmd.exe"}
+  {% else %}
+    {"yes"}
+  {% end %}
+end
+
+private def newline
+  {% if flag?(:win32) %}
+    "\r\n"
+  {% else %}
+    "\n"
+  {% end %}
+end
+
 describe Process do
   it "runs true" do
-    process = Process.new("true")
+    process = Process.new(*exit_code_command(0))
     process.wait.exit_code.should eq(0)
   end
 
   it "runs false" do
-    process = Process.new("false")
+    process = Process.new(*exit_code_command(1))
     process.wait.exit_code.should eq(1)
   end
 
   it "raises if command could not be executed" do
-    expect_raises(RuntimeError, "Error executing process: No such file or directory") do
+    expect_raises(RuntimeError, "Error executing process:") do
       Process.new("foobarbaz", ["foo"])
     end
   end
 
   it "run waits for the process" do
-    Process.run("true").exit_code.should eq(0)
+    Process.run(*exit_code_command(0)).exit_code.should eq(0)
   end
 
   it "runs true in block" do
-    Process.run("true") { }
+    Process.run(*exit_code_command(0)) { }
     $?.exit_code.should eq(0)
   end
 
   it "receives arguments in array" do
-    Process.run("/bin/sh", ["-c", "exit 123"]).exit_code.should eq(123)
+    command, args = exit_code_command(123)
+    Process.run(command, args.to_a).exit_code.should eq(123)
   end
 
   it "receives arguments in tuple" do
-    Process.run("/bin/sh", {"-c", "exit 123"}).exit_code.should eq(123)
+    command, args = exit_code_command(123)
+    Process.run(command, args.as(Tuple)).exit_code.should eq(123)
   end
 
   it "redirects output to /dev/null" do
     # This doesn't test anything but no output should be seen while running tests
-    Process.run("/bin/ls", output: Process::Redirect::Close).exit_code.should eq(0)
+    if {{flag?(:win32)}}
```

Comment:
ditto

Reply:
See https://github.com/crystal-lang/crystal/pull/9047/commits/dffe05a0fee9035ece309238820034b63729cb3d and its CI run.
It doesn't compile like that.
So I did it another way, not sure if it's as nice

Reply:
Looks fine, thank you ❤️